morebits.js

```diff
@@ -3298,6 +3267,63 @@ Morebits.wiki.page = function(pageName, currentAction) {
 
 	};
 
+	// Common checks for action methods
+	// Used for move, undelete, delete, protect, stabilize
+	var fnPreflightChecks = function(action, onFailure) {
+		// if a non-admin tries to do this, don't bother
+		if (!Morebits.userIsSysop && action !== 'move') {
+			ctx.statusElement.error('Cannot ' + action + 'page : only admins can do that');
+			onFailure(this);
+			return false;
+		}
+
+		if (!ctx.editSummary) {
+			ctx.statusElement.error('Internal error: ' + action + ' reason not set (use setEditSummary function)!');
+			onFailure(this);
+			return false;
+		}
+		return true; // all OK
+	};
+
+	// Common checks for fnProcess functions (fnProcessDelete, fnProcessMove, etc.)
+	// Used for move, undelete, delete, protect, stabilize
+	var fnProcessChecks = function(action, onFailure, xml) {
+		var missing = $(xml).find('page').attr('missing') === '';
+		if ((action === 'undelete' && !missing) ||
```

Comment:
This removes the ability to undelete revisions of existing pages. Desirable?

Reply:
Probably not — we had a whole colloquy about that back when I added `undelete`